File: SudokuSolver/Algorithms/CellFinderResult.cs
using SudokuSolver.Domain;

namespace SudokuSolver.Algorithms;

public class CellFinderResult
{
    public Cell Cell { get; set; }
    public int Value { get; set; }

    public bool IsCellFound()
    {
        return Cell != null && Value != 0;
    }

    public static CellFinderResult CellFound(Cell cell, int value)
    {
        return new() { Cell = cell, Value = value };
    }

    public static CellFinderResult CellNotFound()
    {
        return new() { Cell = null, Value = 0 };
    }
}

File: src/main/java/ua/kh/poohdev/sudokusolver/algorithms/CellFinderResult.java
package ua.kh.poohdev.sudokusolver.algorithms;

import ua.kh.poohdev.sudokusolver.domain.Cell;

public class CellFinderResult {

    private final Cell cell;
    private final int value;

    public CellFinderResult(Cell cell, int value) {
        this.cell = cell;
        this.value = value;
    }

    public boolean isCellFound() {
        return cell != null && value != 0;
    }

    public static CellFinderResult notFound() {
        return new CellFinderResult(null, 0);
    }

    public static CellFinderResult found(Cell cell, int value) {
        return new CellFinderResult(cell, value);
    }
}
